Mathlib/Algebra/CharP/Reduced.lean

```diff
@@ -39,12 +47,9 @@ variable {R : Type*} [CommRing R] [IsReduced R]
 @[simp]
 theorem ExpChar.pow_prime_pow_mul_eq_one_iff (p k m : ℕ) [ExpChar R p] (x : R) :
     x ^ (p ^ k * m) = 1 ↔ x ^ m = 1 := by
-  induction' k with k hk
-  · rw [pow_zero, one_mul]
-  · refine' ⟨fun h => _, fun h => _⟩
-    · rw [pow_succ, mul_assoc, pow_mul', ← frobenius_def, ← frobenius_one p] at h
-      exact hk.1 (frobenius_inj R p h)
-    · rw [pow_mul', h, one_pow]
+  rw [pow_mul']
+  convert ← (iterateFrobenius_inj R p k).eq_iff
+  apply map_one
```

Comment:
@riccardobrasca I moved one new lemma and two golfs from [#9311](https://github.com/leanprover-community/mathlib4/pull/9311/files#diff-06dd56536ac06af05c958087a090cbf331a3b2c3e753dd508dabd361e8d7b197) to this PR to streamline [this proof](https://github.com/leanprover-community/mathlib4/pull/9488/commits/e0c5313ee966ce67ddc8ff2d9cd8e6f6fc82baa4#diff-1183eb4a8c4f60106973f5393ebfa817666cb122fe0a95ccdf55c94a4880b965L458-R464) below.